test/README.md

```diff
@@ -0,0 +1,129 @@
+    mms /macro=(V=1) test                          ! OpenVMS
+    nmake V=1 test                                 # Windows
+
+Verbosity on test failure (`VERBOSE_FAILURE` or `VF`, Unix example shown):
+
+    make test VF=1
+
+Verbosity on failed (sub-)tests only (`VERBOSE_FAILURES_ONLY` or `VFO`):
+
+    make test VFO=1
+
+Verbosity on failed (sub-)tests, in addition progress on succeeded (sub-)tests
+(`VERBOSE_FAILURES_PROGRESS` or `VFP`):
+
+    make test VFP=1
+
+If you want to run just one or a few specific tests, you can use
+the `make` variable `TESTS` to specify them, like this:
+
+    make TESTS='test_rsa test_dsa' test            # Unix
+    mms/macro="TESTS=test_rsa test_dsa" test       ! OpenVMS
+    nmake TESTS='test_rsa test_dsa' test           # Windows
+
+And of course, you can combine (Unix examples shown):
+
+    make test TESTS='test_rsa test_dsa' VF=1
+    make test TESTS="test_cmp_*" VFO=1
+
+You can find the list of available tests like this:
+
+    make list-tests                                # Unix
+    mms list-tests                                 ! OpenVMS
+    nmake list-tests                               # Windows
+
+Have a look at the manual for the perl module Test::Harness to
+see what other HARNESS_* variables there are.
+
+To report a bug please open an issue on GitHub, at
+<https://github.com/openssl/openssl/issues>.
+
+For more details on how the `make` variables `TESTS` can be used,
+see section Running Selected Tests below.
+
+Running Selected Tests
+----------------------
+
+The `make` variable `TESTS` supports a versatile set of space separated tokens
+with which you can specify a set of tests to be performed.  With a "current
+set of tests" in mind, initially being empty, here are the possible tokens:
+
+     alltests      The current set of tests becomes the whole set of available
+                   tests (as listed when you do 'make list-tests' or similar).
+
+     xxx           Adds the test 'xxx' to the current set of tests.
+
+    -xxx           Removes 'xxx' from the current set of tests.  If this is the
+                   first token in the list, the current set of tests is first
+                   assigned the whole set of available tests, effectively making
+                   this token equivalent to TESTS="alltests -xxx".
+
+     nn            Adds the test group 'nn' (which is a number) to the current
+                   set of tests.
+
+    -nn            Removes the test group 'nn' from the current set of tests.
+                   If this is the first token in the list, the current set of
+                   tests is first assigned the whole set of available tests,
+                   effectively making this token equivalent to
+                   TESTS="alltests -xxx".
+
+Also, all tokens except for "alltests" may have wildcards, such as *.
+(on Unix and Windows, BSD style wildcards are supported, while on VMS,
+it's VMS style wildcards)
+
+### Examples
```

Comment:
Some markdown renderers will complain if you mix different styles for the headings.

My preference goes for

    # first level heading
    ## second level heading 
    ### third level heading

as I never remember with the alternative "underlining" style which character is used for which heading level

Reply:
Sounds good. Yet when I tried changing to using `#`, `make md-nits` complained:
```
./test/README.md:1: MD003 Header style
./test/README.md:15: MD003 Header style
./test/README.md:67: MD003 Header style
```
So I've kept it as it was because then mdl does not complain and the rendering is fine at least with Firefox.

Reply:
mdl must have been setup explicitly to allow mixing the two styles for 3d level: https://github.com/DavidAnson/markdownlint/blob/HEAD/doc/Rules.md#md003---heading-style

This seems deliberate, so forget my comment!